gulpfile.coffee

```diff
@@ -1,69 +1,214 @@
+dir_storage = "#{dir_temp}/storage"
+
 
-root_dir = './main'
-static_dir = "#{root_dir}/static"
 paths =
+  ext: [
+      "#{dir_ext}/jquery/dist/jquery.js"
+      "#{dir_ext}/moment/moment.js"
+      "#{dir_ext}/nprogress/nprogress.js"
+      "#{dir_ext}/bootstrap-sweetalert/lib/sweet-alert.js"
+      "#{dir_ext}/bootstrap/js/alert.js"
+      "#{dir_ext}/bootstrap/js/button.js"
+      "#{dir_ext}/bootstrap/js/transition.js"
+      "#{dir_ext}/bootstrap/js/collapse.js"
+      "#{dir_ext}/bootstrap/js/dropdown.js"
+      "#{dir_ext}/bootstrap/js/tooltip.js"
+    ]
   clean: [
-      "#{static_dir}/dst"
-      "#{static_dir}/ext"
-      "#{static_dir}/min"
+      dir_min
+      './**/*.pyc'
+      './**/*.pyo'
+      './**/*.~'
+    ]
+  clean_all: [
+      dir_min
+      dir_bower_components
+      dir_node_modules
+      "#{dir_temp}/pip.guard"
+      "#{dir_temp}/npm.guard"
+      "#{dir_temp}/bower.guard"
+    ]
+  clean_py: [
+      dir_lib
+      dir_venv
+    ]
+  style: [
+      "#{dir_style}/style.less"
+    ]
+  script: [
+      "#{dir_script}/**/*.coffee"
     ]
   watch: [
-      "#{static_dir}/**/*.css"
-      "#{static_dir}/**/*.js"
-      "#{root_dir}/**/*.html"
-      "#{root_dir}/**/*.py"
+      "#{dir_static}/**/*.css"
+      "#{dir_static}/**/*.js"
+      "#{dir_main}/**/*.html"
+      "#{dir_main}/**/*.py"
     ]
 
 
-run = (option) ->
-  proc = exec "python -u run.py -#{option}"
-  proc.stderr.on 'data', (data) -> process.stderr.write data
-  proc.stdout.on 'data', (data) -> process.stdout.write data
+gulp.task 'script', false, ->
+  gulp.src(paths.script)
+      .pipe($.sourcemaps.init())
+      .pipe($.coffee().on('error', $.util.log))
+      .pipe($.concat('script.dev.js'))
+      .pipe($.sourcemaps.write())
+      .pipe(gulp.dest("#{dir_min}/script"))
+      .pipe($.rename('script.js'))
+      .pipe($.uglify())
+      .pipe(gulp.dest("#{dir_min}/script"))
+
+
+gulp.task 'ext', false, ->
+  gulp.src(paths.ext)
+      .pipe($.sourcemaps.init())
+      .pipe($.concat('ext.dev.js'))
+      .pipe($.sourcemaps.write())
+      .pipe(gulp.dest("#{dir_min}/script"))
+      .pipe($.rename('ext.js'))
+      .pipe($.uglify())
+      .pipe(gulp.dest("#{dir_min}/script"))
+
+
+gulp.task 'style', false, ->
+  gulp.src(paths.style)
+    .pipe($.sourcemaps.init())
+    .pipe($.less())
+    .pipe($.sourcemaps.write())
+    .pipe($.rename('style.dev.css'))
+    .pipe(gulp.dest("#{dir_min}/style"))
+    .pipe($.rename('style.css'))
+    .pipe($.minifyCss())
+    .pipe(gulp.dest("#{dir_min}/style"))
+
+
+gulp.task 'build',
+"Compiles styles & scripts files into minified version
+and pack python dependencies into #{file_lib}.",
+  $.sequence('clean', 'npm', 'pip', 'ext_install', 'ext', 'pip', ['script', 'style', 'make_lib_zip'])
+
+
+gulp.task 'clean',
+'Clean the project from temp files, python compiled files
+and minified styles and scripts.', ->
+  del paths.clean
+  del paths.clean_py
 
 
-gulp.task 'clean', ->
+gulp.task 'clean_all',
```

Comment:
instead of `clean_all` what about something like `init`?

because basically that's what it does..